app/controllers/schools/charts_controller.rb

```diff
@@ -10,34 +10,43 @@ class Schools::ChartsController < ApplicationController
   before_action :check_aggregated_school_in_cache
 
   def show
-    @chart_type = params.require(:chart_type).to_sym
     respond_to do |format|
-      format.html do
-        set_measurement_options
-        @measurement = measurement_unit(params[:measurement])
-        @title = @chart_type.to_s.humanize
-      end
-      format.json do
-        chart_config = {
-          mpan_mprn: params[:mpan_mprn],
-          series_breakdown: params[:series_breakdown],
-          date_ranges: get_date_ranges
-        }
-        y_axis_units = params[:chart_y_axis_units]
-        chart_config[:y_axis_units] = y_axis_units.to_sym if y_axis_units.present?
-
-        output = ChartData.new(@school, aggregate_school, @chart_type, chart_config, transformations: get_transformations).data
-        if output
-          render json: ChartDataValues.as_chart_json(output)
-        else
-          render json: {}
-        end
-      end
+      format.html { render_html }
+      format.json { render_json }
     end
   end
 
 private
 
+  def render_html
+    @chart_type = params.require(:chart_type).to_sym
```

Comment:
really this should use strong params throughout the controller but i'm a bit reluctant to do a big refactor here without proper test coverage.  e.g.
```
  def chart_params
    params.require(:chart_type).permit(:date_ranges, :transformations, :school_id, :y_axis_units, :chart_y_axis_units, :series_breakdown, :mpan_mprn, :measurement)
  end
```

```
    chart_config = {
      mpan_mprn: chart_params.mpan_mprn,
      series_breakdown: chart_params.series_breakdown,
      date_ranges: get_date_ranges
    }
```